src/bank02.asm

```diff
@@ -8130,29 +8126,197 @@ titleScreenLicenseText:
 ;@ffa_text amount=24
 intoScrollText:
     TXT  "Tree of Mana grows<00>"                      ;; 02:7e8a ...................
-    TXT  "with the energy of<00>"                      ;; 02:7e9d ...................
-    TXT  "will from each and<00>"                      ;; 02:7eb0 ...................
-    TXT  "  every thing of<00>"                        ;; 02:7ec3 .................
-    TXT  "    this world.<00>"                         ;; 02:7ed4 ................
-    TXT  "<00>"                                        ;; 02:7ee4 ?
-    TXT  "  It grows high<00>"                         ;; 02:7ee5 ????????????????
-    TXT  " above the clouds<00>"                       ;; 02:7ef5 ??????????????????
-    TXT  "in the air on top<00>"                       ;; 02:7f07 ??????????????????
-    TXT  "of Mount Illusia.<00>"                       ;; 02:7f19 ??????????????????
-    TXT  "<00>"                                        ;; 02:7f2b ?
-    TXT  "Legend tells that<00>"                       ;; 02:7f2c ??????????????????
-    TXT  "it gives eternal<00>"                        ;; 02:7f3e ?????????????????
-    TXT  "power to the one<00>"                        ;; 02:7f4f ?????????????????
-    TXT  " who touched it.<00>"                        ;; 02:7f60 ?????????????????
-    TXT  "<00>"                                        ;; 02:7f71 ?
-    TXT  "  Dark Lord was<00>"                         ;; 02:7f72 ????????????????
-    TXT  "  trying to find<00>"                        ;; 02:7f82 ????????????????.
-    TXT  "  the way to the<00>"                        ;; 02:7f93 ...........??????
-    TXT  " Tree of Mana to<00>"                        ;; 02:7fa4 ?????????????????
-    TXT  "  get the mighty<00>"                        ;; 02:7fb5 ?????????????????
-    TXT  " power to conquer<00>"                       ;; 02:7fc6 ??????????????????
-    TXT  "    the world.<00>"                          ;; 02:7fd8 ???????????????
-    TXT  "<00>"                                        ;; 02:7fe7 ?
-    db   $00, $00, $00, $00, $00, $00, $00, $00        ;; 02:7fe8 ????????
-    db   $00, $00, $00, $00, $01, $ff, $ff, $ff        ;; 02:7ff0 ????????
-    db   $ff, $ff, $ff, $ff, $ff, $ff, $ff, $ff        ;; 02:7ff8 ????????
+    TXT  "<00>"                                        ;; 02:7fd8 ?
+    db   $01                                           ;; 02:7fd9 ?
+
+; Draws left aligned number HL as modified tiles at WRAM position DE
+; C holds the drawing mode in the first nibble (0 for shift, 1 for shift/swap)
+; C holds number of max digits in the second nibble
+drawLeftAlignedNumberInWRAM:
+    ld   B, $00 ; digit counter
+
+    ; Housekeeping, get a copy of proper WRAM starting
+    ;  location on the stack for later use
+    push BC
+    push DE
+
+    ; Load up the digits on the stack
+.divmod_and_store_number:
+    ld   A, $0a
+    push BC
+    push DE
+    call divMod
```

Comment:
My preference is to optimize for size instead of cycles, but if you want to optimize for cycles, my understanding is that BCD conversion is faster than a `divMod` loop. See `convertToUnpackedBCD` and `convertToPackedBCD` in bank 0 for (confusing) reference.

I also point this out because if money amounts are to be extended to 99999, I think these may be the only functions that handle numbers greater than 65536.

Also, divMod-by-ten loops are not uncommon in this code, so an optimized path for that might be interesting.

Don't get me wrong, though: BCD conversion confuses me.